utils/obsproc/Rads2Ioda.h

```diff
@@ -84,25 +84,25 @@ namespace gdasapp {
       iodaVars.strGlobalAttr_["references"] = references;
 
       // Read optional integer metadata "pass" and "cycle"
-      int pass[iodaVars.location_];  // NOLINT
-      ncFile.getVar("pass").getVar(pass);
-      int cycle[iodaVars.location_];  // NOLINT
-      ncFile.getVar("cycle").getVar(cycle);
+      std::vector<int> pass(iodaVars.location_);
+      ncFile.getVar("pass").getVar(pass.data());
+      std::vector<int> cycle(iodaVars.location_);
+      ncFile.getVar("cycle").getVar(cycle.data());
 
       // Store optional metadata, set ocean basins to -999 for now
       for (int i = 0; i < iodaVars.location_; i++) {
         iodaVars.intMetadata_.row(i) << pass[i], cycle[i], mission_index, -999;
       }
 
       // Get adt_egm2008 obs values and attributes
-      int adt[iodaVars.location_];  // NOLINT
-      ncFile.getVar("adt_egm2008").getVar(adt);
+      std::vector<int> adt(iodaVars.location_);
+      ncFile.getVar("adt_egm2008").getVar(adt.data());
       float scaleFactor;
       ncFile.getVar("adt_egm2008").getAtt("scale_factor").getValues(&scaleFactor);
 
       // Read sla
-      short sla[iodaVars.location_];  // NOLINT
-      ncFile.getVar("sla").getVar(sla);
+      std::vector<int> sla(iodaVars.location_);
```

Comment:
I don't know if it really matters, but the type isn't correct. You might have to use ```int16_t```.
Did you re-run the test?

Reply:
Is a develop branch now writing "sla" to output? I don't see them after converter.
```
(gdasapp) Orion-login-3[54] mindoc$ ncdump -h rads_adt_3b_2021181.ioda.nc
netcdf rads_adt_3b_2021181.ioda {
dimensions:
        Location = 11 ;
variables:
        int Location(Location) ;
                Location:suggested_chunk_dim = 11LL ;

// global attributes:
                string :_ioda_layout = "ObsGroup" ;
                :_ioda_layout_version = 0 ;
                string :obs_source_files = "rads_adt_3b_2021181.nc4" ;
                string :mission_index = "CRYOSAT2 = 6 JASON-1 = 3 JASON-2 = 4 JASON-3 = 5 SARAL = 7 SNTNL-3A = 1 SNTNL-3B = 2 " ;
                string :references = "RADS Data Manual, Version 4.2 or later" ;

group: MetaData {
  variables:
        int cycle(Location) ;
                cycle:_FillValue = -2147483643 ;
        int64 dateTime(Location) ;
                dateTime:_FillValue = -9223372036854775801LL ;
                string dateTime:units = "seconds since 1858-11-17T00:00:00Z" ;
        float latitude(Location) ;
                latitude:_FillValue = -3.368795e+38f ;
        float longitude(Location) ;
                longitude:_FillValue = -3.368795e+38f ;
        float mdt(Location) ;
                mdt:_FillValue = -3.368795e+38f ;
        int mission(Location) ;
                mission:_FillValue = -2147483643 ;
        int pass(Location) ;
                pass:_FillValue = -2147483643 ;
  } // group MetaData

group: ObsError {
  variables:
        float absoluteDynamicTopography(Location) ;
                absoluteDynamicTopography:_FillValue = -3.368795e+38f ;
  } // group ObsError

group: ObsValue {
  variables:
        float absoluteDynamicTopography(Location) ;
                absoluteDynamicTopography:_FillValue = -3.368795e+38f ;
  } // group ObsValue

group: PreQC {
  variables:
        int absoluteDynamicTopography(Location) ;
                absoluteDynamicTopography:_FillValue = -2147483643 ;
  } // group PreQC
}
```
I will reverse ```int``` to ```short``` since using ```short``` can conserve memory when we using a large array.

Reply:
no @apchoiCMD , it's writing MDT in the metadata.

Reply:
@guillaumevernieres Thanks for confirming this!